### PR TITLE
Add Monad Devnet support

### PR DESCRIPTION
**Pull request by bruce-riley**
This PR adds support for the Monad Devnet environment. This is being added separately from regular Monad testnet (TBD) because it is expected that this environment will be upgraded and reset more often than a standard testnet environment.

This PR only adds support for the core contract, which is available [here](https://brightstar-884.devnet1.monad.xyz/address/0x376428e7f26D5867e69201b275553C45B09EE090).

This PR does not specify an RPC endpoint because there is not currently a public one available.

### CODE REVIEW

File: core/base/src/constants/finality.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ const finalityThresholds = [
   ["BaseSepolia", 512],
   ["OptimismSepolia", 512],
   ["PolygonSepolia", 32],
+  ["MonadDevnet", 0],
```
Comment (evan-gray) [inline core/base/src/constants/finality.ts:79]: should this be a 1 like Fantom?

File: core/base/src/constants/rpc.ts [hunk 2]
```diff
@@ -91,6 +91,7 @@ const rpcConfig = [[
     ["Snaxchain",       "https://testnet.snaxchain.io"],
     ["Unichain",        "https://sepolia.unichain.org"],
     ["Worldchain",      "https://worldchain-sepolia.g.alchemy.com/public"],
+    ["MonadDevnet",     "TBD"],
```
Comment (evan-gray) [inline core/base/src/constants/rpc.ts:94]: oof. what would @kev1n-peters recommend here?

Comment (kev1n-peters) [reply core/base/src/constants/rpc.ts:94]: Maybe we could set it to the empty string and add a comment saying, "TODO: no public rpc is currently available, override with a custom rpc for now"

File: core/base/src/constants/finality.ts (hunk 1 above)
Comment (bruce-riley) [reply core/base/src/constants/finality.ts:79]: I changed MonadDevent and Berachain to 1 and SnaxChain, Unichain and Worldchain to 512 per offline discussion with @evan-gray.

Comment (evan-gray) [reply core/base/src/constants/finality.ts]: concur, this is just to match the other OP Stack chains, not because it's factually accurate 👀

**Review by evan-gray — APPROVED** [review]

**Review by kev1n-peters — APPROVED** [review]

**Pull request merged**